packages/lodestar/src/api/impl/beacon/blocks/utils.ts

```diff
@@ -36,7 +36,7 @@ export async function resolveBlockId(
     return db.blockArchive.get(GENESIS_SLOT);
   }
   if (blockId === "finalized") {
-    return db.block.get(forkChoice.getFinalizedCheckpoint().root.valueOf() as Uint8Array);
+    return await db.blockArchive.get(forkChoice.getFinalizedCheckpoint().epoch);
```

Comment:
don't we have last finalized block in memory?

Reply:
I don't think so, I don't think we have any blocks in memory for much time. Either pushed into the `block`, `pendingBlock` or `blockArchive` dbs.